src/Store/SyntaxHighlightingStoreConnector.re

```diff
@@ -43,6 +42,46 @@ let start = (languageInfo: Ext.LanguageInfo.t) => {
         effect,
       );
 
+  let syntaxGrammarRepository =
+    Oni_Syntax.GrammarRepository.create(languageInfo);
+
+  let getEagerLines = (~scope, ~configuration, ~theme, lines) => {
+    let maxLines =
+      configuration |> Core.Configuration.getValue(c => c.syntaxEagerMaxLines);
+    let maxLineLength =
+      configuration
+      |> Core.Configuration.getValue(c => c.syntaxEagerMaxLineLength);
+
+    let len = min(Array.length(lines), maxLines);
+    let idx = ref(0);
+    let limitExceeded = ref(false);
+
+    while (idx^ < len && ! limitExceeded^) {
+      if (String.length(lines[idx^]) > maxLineLength) {
+        limitExceeded := true;
+      };
+
+      incr(idx);
+    };
+
+    let numberOfLinesToHighlight = idx^;
```

Comment:
The idiomatic way of doing this is to use a for loop and an exception for the early termination. No refs needed, although an exception certainly isn't pure either. The purely functional approach would be to use a recursive helper function.

Reply:
Switched to a recursive helper function in latest commit